starter-code/index.js

```diff
@@ -1,11 +1,49 @@
 class SortedList {
-  constructor() {}
-  add(item) {}
-  get(pos) {}
-  max() {}
-  min() {}
-  average() {}
-  sum() {}
+  constructor() {
+    this.items = [];
+    this.length = 0;
+  }
+  add(x) {
+    this.items.push(x);
+    this.items.sort( function (a, b) {
+      return a - b;
+    });
+    this.length++;
+  }
+
+  get(pos) {
+    return this.items[pos -1];
+  }
+
+  max() {
+    if (this.items.length === 0) {
+      throw new Error("EmptySortedList");
+    }
+    return this.items[this.items.length - 1];
+  }
+
+  min() {
+    if (this.items.length === 0) {
+      throw new Error("EmptySortedList");
+    }
+    return this.items[0];
+  }
+
+  sum() {
+    if (this.items.length === 0) {
+      return 0;
+    }
+    return this.items.reduce((acc, oneItem) => acc += oneItem, 0);
+  }
+
+  average() {
+    if (this.items.length === 0) {
```

Comment:
when calculating the average, you can call the sum function. this will help keep your code dry

```
    return this.sum() / this.items.length;

```